changelog/4562.deprecation.rst

```diff
@@ -0,0 +1,4 @@
+Deprecate configuring hook specs/impls using attributes/marks.
```

Comment:
I think it would be good to list the marks/attributes, so people reading can do a grep search.

Reply:
good point, will put it into the deprecation docs